crates/sui-adapter/src/adapter.rs

```diff
@@ -798,6 +805,13 @@ fn handle_transfer<
     // thus, all objects in the global object pool have version > 0
     move_obj.increment_version();
     let obj_id = move_obj.id();
+
+    // update the shared object initial version, if applicable
+    if let Owner::Shared { shared_at_version } = &mut recipient {
+        debug_assert!(shared_at_version.value() == 0);
```

Comment:
same here - `assert!` please.

Reply:
I would disagree for the same reason, more of a precondition than an actual assert